apps/src/applab/firebaseUtils.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+/* global Applab */
+
+var Firebase = require("firebase");
+
+/**
+ * Namespace for Firebase utility functions.
+ */
+var FirebaseUtils = module.exports;
+
+var databaseCache = {};
+FirebaseUtils.getDatabase = function (channelId) {
+  var db = databaseCache[channelId];
+  if (!db) {
+    if (!Applab.firebaseName) {
+      throw new Error("Error connecting to Firebase: Firebase name not specified");
+    }
+    if (!Applab.firebaseAuthToken) {
+      throw new Error("Error connecting to Firebase: Firebase auth token not specified");
+    }
+    var base_url = 'https://' + Applab.firebaseName + '.firebaseio.com';
+    db = new Firebase(base_url + '/v3/channels/' + channelId);
+    if (Applab.firebaseAuthToken) {
+      db.authWithCustomToken(Applab.firebaseAuthToken, function (err, user) {
+        if (err) {
+          throw new Error('error authenticating to Firebase: ' + err);
+        } else {
+          Applab.firebaseUserId = user.uid;
+        }
+      });
+    }
+    databaseCache[channelId] = db;
+  }
+  return db;
+};
```

Comment:
The things that I think will have the most value (in order):
(1) arrow functions, i.e. instead
```
db.authWithCustomToken(token, function (err, user) {
  ...
})
// do
db.authWIthCustomToken(token, (err, user) => {
...
});
// not a big difference here, but can be much more readable for short functions imo
```
(2) import/export instead of require/module.exports =

(3) let/const instead of var

(4) backticks for strings with variables, i.e.
```
const basel_url = `https://${Applab.firebasename}.firebaseio.com`;
```

None of this needs to block checkin, but prob good to start familiarizing yourself with/using going forward. Let me know if you have any questions :)